tensorflow/python/ops/signal/fft_ops.py

```diff
@@ -358,11 +358,11 @@ def fftshift(x, axes=None, name=None):
     x = _ops.convert_to_tensor(x)
     if axes is None:
       axes = tuple(range(x.shape.ndims))
-      shift = [int(dim // 2) for dim in x.shape]
+      shift = [dim // 2 for dim in _array_ops.shape(x)]
     elif isinstance(axes, int):
-      shift = int(x.shape[axes] // 2)
+      shift = _array_ops.shape(x)[axes] // 2
     else:
-      shift = [int((x.shape[ax]) // 2) for ax in axes]
+      shift = [(_array_ops.shape(x)[ax]) // 2 for ax in axes]
```

Comment:
This will avoid creating a shape op for each ax:
```
shift = _array_ops.gather(_array_ops.shape(x), axes) // 2
```

Reply:
right, thanks!